src/lib/y2storage/proposal/partitions_distribution_calculator.rb

```diff
@@ -326,6 +331,124 @@ def add_unused_spaces(dist_hashes, spaces)
         spaces_hash = Hash[spaces.product([[]])]
         dist_hashes.map! { |d| spaces_hash.merge(d) }
       end
+
+      # Based on the partition to be resized, sets the FreeDiskSpace#growing? attribute
+      # in one of the existing spaces, or adds a new FreeDiskSpace object to the
+      # collection if a new space will be created.
+      #
+      # @see resizing_size
+      # @see FreeDiskSpace#growing?
+      #
+      # @param free_spaces [Array<FreeDiskSpace>] initial list of free spaces in
+      #   the system (#growing? returns false for all of them)
+      # @param partition [Partition] partition to be resized
+      # @return [Array<FreeDiskSpace>] list of free spaces containing a growing one
+      #   (added to the list or replacing the original space affected by the resizing)
+      def add_or_mark_growing_space(free_spaces, partition)
```

Comment:
As commented offline, maybe a comment the case where the next free space is the first region of an extended partition. And a test case?